slider/index.js

```diff
@@ -0,0 +1,277 @@
+        swipeContainer.appendChild(firstClone); // Add the first clone at the end
+        swipeContainer.insertBefore(lastClone, slides[0]); // Add the last clone at the beginning
+    }
+
+    /**
+     * Update the total number of slides
+     * 
+     * @param {number} totalSlides - The total number of slides
+     * @returns {void}
+     */
+    updateTotalSlides(totalSlides) {
+        this.querySelector(this.CLASSNAMES.TOTAL_SLIDES).textContent = totalSlides;
+    }
+
+    /**
+     * Initialize the slider position
+     * 
+     * @param {HTMLElement} swipeContainer - The swipe container element
+     * @returns {void}
+     */
+    initializePosition(swipeContainer) {
+        this.currentSlide = 1;
+        swipeContainer.style.transform = `translateX(-${this.currentSlide * 100}%)`;
+    }
+
+    /**
+     * Add event listeners for navigation and touch/drag
+     * Passing the total slides count to update the pagination separately
+     * 
+     * @param {HTMLElement} swipeContainer - The swipe container element
+     * @param {number} totalSlides - The total number of slides
+     * 
+     * @returns {void}
+     */
+    addEventListeners(swipeContainer, totalSlides) {
+        const clonedSlidesCount = swipeContainer.querySelectorAll(this.CLASSNAMES.SLIDES).length;
+
+        const updateSlides = () => this.updateSlides(swipeContainer, clonedSlidesCount, totalSlides);
+        const resetPosition = () => this.resetPosition(swipeContainer, clonedSlidesCount, totalSlides);
+        const nextSlide = () => this.moveToNextSlide(updateSlides, clonedSlidesCount);
+        const prevSlide = () => this.moveToPrevSlide(updateSlides, clonedSlidesCount);
+
+        this.setupDragAndDrop(swipeContainer, nextSlide, prevSlide, updateSlides);
+
+        this.addListener(this.querySelector(this.CLASSNAMES.PREV_BUTTON), 'click', prevSlide);
+        this.addListener(this.querySelector(this.CLASSNAMES.NEXT_BUTTON), 'click', nextSlide);
+
+        this.addListener(swipeContainer, 'transitionend', resetPosition);
+    }
+
+    /**
+     * Add the event listener globally to be used across component
+     * It will be removed in disconnectedCallback
+     * 
+     * @param {HTMLElement} element - The element to add the event listener to
+     * @param {string} event - The event to listen for
+     * @param {Function} handler - The event handler
+     * 
+     * @returns {void}
+     */
+    addListener(element, event, handler) {
+        if (element) {
+            element.addEventListener(event, handler);
+            this.eventListeners.push({ element, event, handler });
+        }
+    }
+
+    /**
+     * Update slides position and pagination
+     */
+    updateSlides(swipeContainer, clonedSlidesCount, totalSlides) {
+        swipeContainer.style.transition = this.transition;
+        swipeContainer.style.transform = `translateX(-${this.currentSlide * 100}%)`;
+
+        const displayedSlideIndex =
```

Comment:
@samuelsilvapt it's quite difficult to read nested ternary statements. To me making it a bit more verbose really helps to be able to easily follow the logic. Something like:

```
let displayedSlideIndex;

if (this.currentSlide === 0) {
    displayedSlideIndex = totalSlides;
    return displayedSlideIndex;
}

if (this.currentSlide === clonedSlidesCount - 1) {
    displayedSlideIndex = 1;
    return displayedSlideIndex;
}

displayedSlideIndex = this.currentSlide;
return displayedSlideIndex;
```

Not as sexy, but easier for me to process.

Reply:
@ejcromwell I've replaced this with a switch case